repose-aggregator/components/services/atom-feed-service/atom-feed-service-impl/src/main/scala/org/openrepose/nodeservice/atomfeed/impl/AtomEntryStreamBuilder.scala

```diff
@@ -112,6 +132,10 @@ object AtomEntryStreamBuilder {
 
   case class UnrecoverableIOException(cause: Throwable) extends RuntimeException(cause)
 
+  case class ServerResponseException(message: String) extends Exception(message)
+
+  object ClientErrorException extends Exception
+
   object AuthenticationException extends Exception
```

Comment:
Not sure what you mean by full inner classes, but just because the only usages right now have that doesn't mean that someone who picks it up in the future will do the right thing. I always assume that i'm not going to be as careful later as i am being right now. Messages cost us basically nothing to put on them, so message them up.